R/compound_taskid-utils.R

```diff
@@ -0,0 +1,202 @@
+#' Detect the compound_taskid_set for a tbl for each modeling task in a given round.
+#'
+#' @param tbl a tibble/data.frame of the contents of the file being validated.
+#' Column types must **all be character**.
+#' @param config_tasks a list representantion of the `tasks.json` config file.
+#' @param round_id Character string. The round ID.
+#' @param compact Logical. If TRUE, the output will be compacted to remove NULL elements.
+#' @param error Logical. If TRUE, an error will be thrown if the compound task ID set is not valid.
+#' If FALSE and an error is detected, the detected compound task ID set will be
+#' returned with error attributes attached.
+#'
+#' @return A list of vectors of compound task IDs detected in the tbl, one for each
+#' modeling task in the round. If `compact` is TRUE, modeling tasks returning NULL
+#' elements will be removed.
+#' @noRd
+get_tbl_compound_taskid_set <- function(tbl, config_tasks, round_id,
+                                        compact = TRUE, error = TRUE) {
+  tbl <- tbl[tbl$output_type == "sample", names(tbl) != "value"]
+  out_tid <- hubUtils::std_colnames["output_type_id"]
+
+  mt_compound_taskids <- get_round_compound_task_ids(
+    config_tasks,
+    round_id
+  )
+
+  mt_tbls <- purrr::map(
+    .x = expand_model_out_grid(
+      config_tasks = config_tasks,
+      round_id = round_id,
+      all_character = TRUE,
+      include_sample_ids = FALSE,
+      bind_model_tasks = FALSE
+    ),
+    function(.x) {
+      if (!has_spls_tbl(.x)) {
+        return(NULL)
+      }
```

Comment:
This comment may fall under the category of premature optimization, but given recent discussion about how expand-gridding can be costly -- could we consider adding an argument to `expand_model_out_grid` to only do this for groups that have samples as an output type?

Reply:
That's the plan! See https://github.com/hubverse-org/hubValidations/issues/98